### PR TITLE
Create file modules

### PR DESCRIPTION
**Pull request by etiennenoel**

### CODE REVIEW

File: packages/file/src/managers/directory.manager.ts [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            if (options.match) {
+                let target;
+                if (options.matchType === MatchTypeEnum.Extension) {
+                    target = ext;
+                } else if (options.matchType === MatchTypeEnum.Base) {
+                    target = basename;
+                } else if (options.matchType === MatchTypeEnum.Filename) {
+                    target = filename;
+                } else {
+                    throw new Error(`options.matchType contains invalid value "${options.matchType}", should be "MatchTypeEnum.Extension", "MatchTypeEnum.Base" or "MatchTypeEnum.Filename"`);
```
Comment (mathieugh) [inline packages/file/src/managers/directory.manager.ts:63]: Here I'd add some context to the error message to make it clear it's happened during listing the files

File: packages/file/src/managers/directory.manager.ts [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+            if (options.resultType === DirectoryListResultEnum.FileInfoObject) {
+                options.results.push(obj);
+            } else {
+                options.results.push(fullPath);
+            }
+
+        }
+        if (options.recurse) {
+            options.skipTopLevelFiles = false;        // turn this off for recursion
+            for (const d of dirs) {
```
Comment (mathieugh) [inline packages/file/src/managers/directory.manager.ts:100]: Maybe replace d for dir ?

File: packages/file/src/managers/directory.manager.ts [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+            resultType: DirectoryListResultEnum.FilePath,
+            results: [],
+        };
+
+        if(options !== undefined) {
+            if (typeof options.match === "string") {
+                defaults.matchCaseInsensitive = true;
+            } else {
+                if (options.matchCaseInsensitive) {
+                    throw new TypeError("options.matchCaseInsensitive can only be specified when options.match is a string");
```
Comment (mathieugh) [inline packages/file/src/managers/directory.manager.ts:140]: Same thing here

File: packages/file/src/options/directory-list.options.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import {MatchTypeEnum} from "../enums/match-type.enum";
+import {TypesEnum} from "../enums/types.enum";
+import {FileInfoInterface} from "../interfaces/file-info.interface";
+import {DirectoryListResultEnum} from "../enums/directory-list-result.enum";
+
+export interface DirectoryListOptions {
+    matchType?: MatchTypeEnum;
```
Comment (mathieugh) [inline packages/file/src/options/directory-list.options.ts:7]: Missing comment here, we should specify the difference between matchType and types

**Review by mathieugh — CHANGES_REQUESTED** [review]

File: packages/file/src/managers/directory.manager.ts (hunk 2 above)
Comment (etiennenoel) [reply packages/file/src/managers/directory.manager.ts:100]: Already have a dir variable

File: packages/file/src/managers/directory.manager.ts [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+        // make copy of options object and initialize defaults
+        // we put results array in the options so it can be passed into the recursive calls and
+        //    they can just add to the same array rather than having to merge arrays
+        const defaults: DirectoryListOptions = {
+            matchType: MatchTypeEnum.Extension,
+            types: TypesEnum.File,
+            resultType: DirectoryListResultEnum.FilePath,
+            results: [],
+        };ù
```
Comment (mathieugh) [inline packages/file/src/managers/directory.manager.ts:133]:
```suggestion
        };
```

**Review by mathieugh — CHANGES_REQUESTED** [review]

**Pull request merged**